controllers/twophase/types.go

```diff
@@ -92,15 +92,50 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 
 		status.Experiment.Phase = v1alpha1.ExperimentPhaseFinished
 		status.FailedMessage = emptyString
-	} else if chaos.IsPaused() {
+	} else if chaos.GetPause() != "" {
 		if status.Experiment.Phase == v1alpha1.ExperimentPhaseRunning {
 			r.Log.Info("Pausing")
 
-			err = r.Recover(ctx, req, chaos)
-			if err != nil {
-				r.Log.Error(err, "failed to pause chaos")
-				updateFailedMessage(ctx, r, chaos, err.Error())
-				return ctrl.Result{Requeue: true}, err
+			if chaos.GetPause() == "true" {
+				err = r.Recover(ctx, req, chaos)
+				if err != nil {
+					r.Log.Error(err, "failed to pause chaos")
+					updateFailedMessage(ctx, r, chaos, err.Error())
+					return ctrl.Result{Requeue: true}, err
+				}
+			} else {
+				// Pause time is set
+				var pauseSpec v1alpha1.SchedulerSpec
+				pauseSpec.Cron = "every " + chaos.GetPause()
+				nextStart, err := utils.NextTime(pauseSpec, time.Now())
+				if err != nil {
+					r.Log.Error(err, "failed to get the next start time after pause, check the format of pause input")
+					return ctrl.Result{}, err
+				}
+
+				err = r.Recover(ctx, req, chaos)
+				if err != nil {
+					r.Log.Error(err, "failed to pause chaos")
+					updateFailedMessage(ctx, r, chaos, err.Error())
+					return ctrl.Result{Requeue: true}, err
+				}
+
+				if nextStart.Before(chaos.GetNextRecover()) {
+					chaos.SetNextStart(*nextStart)
+				} else {
+					cronGap := getCronGap(ctx, r, chaos)
```

Comment:
I didn't understand this logic here. Why we can't use the following code directly? 
```
chaos.SetNextStart(nextStart)
chaos.SetNextRecover(nextStart.Add(duration)
```

Reply:
We want to keep in the same pattern as cron told us. 
So if the resume time should be in waiting state, it would change status from pausing to waiting, and wait for the next start, rather than start immediately.